File: javascript/sdk/wallet.js
import { PWRWallet } from "@pwrjs/core";
const randomWallet = new PWRWallet();

// Get the wallet address
const address = randomWallet.getAddress();
console.log(`Address: ${address}`);

// Get the wallet's private key
// const privateKey = randomWallet.getPrivateKey();
// console.log(`PrivateKey: ${privateKey}`);

// Get the wallet balance
randomWallet.getBalance()
    .then(balance => console.log(`Balance: ${balance}`));

// Get the wallet's current nonce
randomWallet.getNonce()
    .then(nonce => console.log(`Nonce: ${nonce}`));

// Create a wallet from an existing private key (String || ByteArray || Int)
// in this example we will store the private key as a string
const privateKey = "0x04828e90065864c111871769c601d7de2246570b39dd37c19ccac16c14b18f72";
const wallet = new PWRWallet(privateKey);
console.log(wallet.getAddress());

File: rust/examples/sdk/wallet.rs
use pwr_rs::Wallet;

#[tokio::main]
pub async fn main() {
    let random_wallet = Wallet::new_random(12);

    // Get the wallet address
    let address = random_wallet.get_address();
    println!("Address: {address}");

    // Get the wallet's private key
    let seed_phrase = random_wallet.get_seed_phrase();
    println!("Seed Phrase: {seed_phrase}");

    // Get the wallet's public key
    let public_key = random_wallet.get_public_key();
    println!("Public Key: {:?}", hex::encode(public_key));

    // Get the wallet's private key
    let private_key = random_wallet.get_private_key();
    println!("Private Key: {:?}", hex::encode(private_key));

    // Get the wallet balance
    let balance = random_wallet.get_balance().await;
    println!("Balance: {balance}");

    // Get the wallet's current nonce
    let nonce = random_wallet.get_nonce().await;
    println!("Nonce: {nonce}");

    // Create a wallet from an existing seed phrase (String)
    // in this example we will store the seed phrase
    let seed_phrase = "badge drive deputy afraid siren always green about certain stuff play surround";
    let wallet = Wallet::new(seed_phrase);

    // Get the wallet's address
    let address = wallet.get_address();
    println!("Address: {address}");
}
